### PR TITLE
fix(parser/binder): remove/reject not supported stmt

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
close https://github.com/risingwavelabs/risingwave/issues/4315
## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by neverchanje — APPROVED** [review]
Agree with the motivation of this PR.

File: src/sqlparser/src/ast/mod.rs [hunk 1]
```diff
@@ -1155,13 +1137,11 @@ impl fmt::Display for Statement {
                 name,
                 columns,
                 constraints,
-                table_properties,
                 with_options,
```
Comment (BugenZhao) [inline src/sqlparser/src/ast/mod.rs:1140]: Oops. How does the parser distinguish them?

**Review by BugenZhao — APPROVED** [review]
LGTM

File: src/sqlparser/src/ast/mod.rs (hunk 1 above)
Comment (xxchan) [reply src/sqlparser/src/ast/mod.rs:1140]:
```rust
        let with_options = self.parse_with_properties()?;
        let table_properties = self.parse_options(Keyword::TBLPROPERTIES)?;
```

**Pull request merged**